imenu-list.el

```diff
@@ -270,9 +270,10 @@ Each entry is appended to `imenu-list--line-entries' as well
  (`imenu-list--line-entries' is cleared in the beginning of this
 function)."
   (read-only-mode -1)
-  (erase-buffer)
+  (delete-region (point-min) (point-max))
```

Comment:
Why replace `erase-buffer` with `delete-region`? The only thing `erase-buffer` adds is that it calls `Fwiden`, in C, so it is very fast (constant overhead). Using `delete-region` like this also does not properly delete the imenu-list if it is narrowed.

Reply:
I don't remember why I did that, this PR has been around for so long. I've changed it back now.